### PR TITLE
Add user, get users

### PR DESCRIPTION
**Pull request by stetskivostap**

### CODE REVIEW

File: BLL/Mappings/UserListProfiler.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+using DAL.Entities;
+using System.Collections.Generic;
+
+namespace BLL.Mappings
+{
+    public class UserListProfiler : Profile
+    {
+        public UserListProfiler()
+        {
+            CreateMap<IEnumerable<User>, IEnumerable<UserListDto>>();
```
Comment (vladpavlyuk) [inline BLL/Mappings/UserListProfiler.cs:12]: Don’t write mapper config for collections. You can write config for object and then use it like that`mapper.Map<IEnumerable<UserDto>>(data)`. Automapper will understand how to map collection if it has config for object

File: BLL/Mappings/UserProfiler.cs [hunk 2]
```diff
@@ -9,6 +9,7 @@ public class UserProfiler : Profile
         public UserProfiler()
         {
             CreateMap<UserDto, User>();
+
```
Comment (vladpavlyuk) [inline BLL/Mappings/UserProfiler.cs:12]: Avoid changes like that. In order to make cosmetic changes create another pull request

File: BLL/Services/AuthenticationService.cs [hunk 3]
```diff
@@ -22,13 +23,16 @@ public class AuthenticationService : IAuthenticationService
         private readonly TokenManagement _tokenManagement;
         private readonly IUnitOfWork _unitOfWork;
         private readonly IMapper _mapper;
+        private readonly IRepository<User> _userRepository;
 
         public AuthenticationService(IOptions<TokenManagement> tokenManagement, IUnitOfWork uow,
             IMapper mapper)
         {
+
```
Comment (vladpavlyuk) [inline BLL/Services/AuthenticationService.cs:31]: The same

File: BLL/Services/AuthenticationService.cs [hunk 4]
```diff
@@ -22,13 +23,16 @@ public class AuthenticationService : IAuthenticationService
...
+        private readonly IRepository<User> _userRepository;
 
         public AuthenticationService(IOptions<TokenManagement> tokenManagement, IUnitOfWork uow,
             IMapper mapper)
         {
+            
             _tokenManagement = tokenManagement.Value;
             _unitOfWork = uow;
             _mapper = mapper;
+            _userRepository = _unitOfWork.GetRepository<User>();
```
Comment (vladpavlyuk) [inline BLL/Services/AuthenticationService.cs:35]: Can you inject repository object not unit of work?

File: BLL/Services/AuthenticationService.cs [hunk 5]
```diff
@@ -54,23 +58,27 @@ private string GenerateJwt(User user)
...
+                }
             }
         }
 
 
 
         private User Authorize(string email, string password)
         {
-            var user = _unitOfWork.GetRepository<User>().GetWithInclude(p => p.Role).SingleOrDefault(e => e.Email == email);
+            var user = _userRepository.GetWithInclude(p => p.Role).SingleOrDefault(e => e.Email == email);
```
Comment (vladpavlyuk) [inline BLL/Services/AuthenticationService.cs:81]: Use FirstOrDefault instead of SingleOrDefault throughout the application. Differences between them is here https://www.c-sharpcorner.com/blogs/singleordefault-vs-firstordefault-in-linq-query1

File: BLL/Services/UserService.cs [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        private readonly IRepository<User> _userRepository;
+        public UserService(IUnitOfWork unitOfWork, IMapper mapper)
+        {
+            _unitOfWork = unitOfWork;
+            _userRepository = _unitOfWork.GetRepository<User>();
+                     
+        }
+        public  IEnumerable<User> GetAllUser(PaginationParams pagination)
+        {
+            var users = _userRepository.GetAll();
```
Comment (vladpavlyuk) [inline BLL/Services/UserService.cs:24]: It seems like you can remove users field and just return the result immediately

File: Common/Constants/PaginationParams.cs [hunk 7]
```diff
@@ -0,0 +1,25 @@
+﻿namespace Common.Constants
+{
+    public class PaginationParams
```
Comment (vladpavlyuk) [inline Common/Constants/PaginationParams.cs:3]: I don’t like this class. I see that Skip and Take fields are never used. Just make PageSize and PageNumber as constants and remove the rest

File: VirtualClassRoom/ClientApp/src/app/Models/User.ts [hunk 8]
```diff
@@ -0,0 +1,12 @@
+export interface User{
+    id?: string,
+    email: string,
+    passwordHash: string,
```
Comment (vladpavlyuk) [inline VirtualClassRoom/ClientApp/src/app/Models/User.ts:4]: Remove this. It’s forbidden to return the hash of the password. It’s the sensitive information

File: VirtualClassRoom/ClientApp/src/app/components/add-user/add-user.component.html [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+         placeholder="city" formControlName="city" required>
+      </div>
+
+      <div class="form-control">        
+        <input type="text"
+         placeholder="address" formControlName="address" required>
+      </div>
+
+      <mat-dialog-actions>
+              <button class="btn" (click)="save()" [mat-dialog-close]=""
```
Comment (vladpavlyuk) [inline VirtualClassRoom/ClientApp/src/app/components/add-user/add-user.component.html:44]: Why here is the both save and submit methods? It’s strange

File: BLL/Services/AuthenticationService.cs (hunk 4 above)
Comment (stetskivostap) [reply BLL/Services/AuthenticationService.cs:35]: I wil find out it

File: BLL/Services/UserService.cs [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+    {
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IRepository<User> _userRepository;
+        public UserService(IUnitOfWork unitOfWork, IMapper mapper)
+        {
+            _unitOfWork = unitOfWork;
+            _userRepository = _unitOfWork.GetRepository<User>();
+                     
+        }
+        public  IEnumerable<User> GetAllUser(PaginationParams pagination)
```
Comment (vladpavlyuk) [inline BLL/Services/UserService.cs:22]:
```suggestion
        public  IEnumerable<User> GetAllUser(PaginationParams pagination) => _userRepository.GetAll();
```

File: VirtualClassRoom/ClientApp/src/app/components/add-user/add-user.component.html (hunk 9 above)
Comment (vladpavlyuk) [inline VirtualClassRoom/ClientApp/src/app/components/add-user/add-user.component.html:44]: Leave only the one click event on button

File: VirtualClassRoom/ClientApp/src/app/components/add-user/add-user.component.ts [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+import { User } from 'src/app/Models/User';
+import { Router } from '@angular/router';
+import { UserService } from 'src/app/services/UserService';
+
+@Component({
+  selector: 'app-add-user',
+  templateUrl: './add-user.component.html',
+  styleUrls: ['./add-user.component.css']
+})
+export class AddUserComponent implements OnInit {
```
Comment (vladpavlyuk) [inline VirtualClassRoom/ClientApp/src/app/components/add-user/add-user.component.ts:13]: Implement Add User functionality in the new page and after successful adding navigate to '/users' with reload

File: VirtualClassRoom/ClientApp/src/app/components/user/user.component.ts [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+
+@Component({
+  selector: 'app-user',
+  templateUrl: './user.component.html',
+  styleUrls: ['./user.component.css']
+})
+export class UserComponent {
+  columnsToDisplay = ['email','firstname', 'lastname', 'city', 'phonenumber', 'birthday']  
+  users: User[] = []
+  loading = false
```
Comment (vladpavlyuk) [inline VirtualClassRoom/ClientApp/src/app/components/user/user.component.ts:16]: Use this as loader https://www.npmjs.com/package/ngx-ui-loader Getting started info is here https://tdev.app/ngx-ui-loader
